x-pack/legacy/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -226,16 +226,25 @@ function isBeatFromAPM(bucket) {
 }
 
 async function hasNecessaryPermissions(req) {
-  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('data');
-  const response = await callWithRequest(req, 'transport.request', {
-    method: 'POST',
-    path: '/_security/user/_has_privileges',
-    body: {
-      cluster: ['monitor'],
-    },
-  });
-  // If there is some problem, assume they do not have access
-  return get(response, 'has_all_requested', false);
+  try {
+    const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('data');
+    const response = await callWithRequest(req, 'transport.request', {
+      method: 'POST',
+      path: '/_security/user/_has_privileges',
+      body: {
+        cluster: ['monitor'],
+      },
+    });
+    // If there is some problem, assume they do not have access
+    return get(response, 'has_all_requested', false);
+  } catch (err) {
+    if (
+      err.message === 'no handler found for uri [/_security/user/_has_privileges] and method [POST]'
```

Comment:
@kobelb I know we discussed this awhile back, but I'm revisiting this PR and wondering if you think this check is sufficient enough. We can either do this, or do a pre-flight check for the security feature on the cluster.

Reply:
We've had to do similar checks elsewhere. It's not perfect, but it generally works. I would however encourage you to write an integration test against Elasticsearch to ensure that if they change their error messages, that the test begins to fail. These error messages are prone to change.